src/factorization.jl

```diff
@@ -121,3 +117,23 @@ function restarted_arnoldi(A::AbstractMatrix{T}, min = 5, max = 30, restarts = 4
 
     return λs, view(arnoldi.V, :, 1 : max) * xs
 end
+
+"""
+Run IRAM until the eigenpair is a good enough approximation
+"""
+function restarted_arnoldi_2(A::AbstractMatrix{T}, min = 5, max = 30, criterion = 1e-5) where {T}
+    n = size(A, 1)
+
+    arnoldi = initialize(T, n, max)
+    iterate_arnoldi!(A, arnoldi, 1 : max)
+
+    λs, xs = eig(view(arnoldi.H, 1 : max, 1 : max))
+
+    while abs(arnoldi.H[max+1,max]) * abs(xs[max,1]) > criterion #doesn't seem to be working right, probably because eigenvectors are not sorted
+        implicit_restart!(arnoldi, min, max)
+        iterate_arnoldi!(A, arnoldi, min + 1 : max)
+        λs, xs = eig(view(arnoldi.H, 1 : max, 1 : max)) #and this seems excessive
```

Comment:
Yup, we don't need it at all, since `λs` and `xs` will just be a subset of what we already had.